internal/handlers/pg/pgdb/query.go

```diff
@@ -268,61 +267,133 @@ func prepareWhereClause(sqlFilters *types.Document) (string, []any, error) {
 			}
 		}
 
-		// Handle _id with a simpler query, as it can't be an array
-		if k == "_id" {
-			switch v := v.(type) {
-			case *types.Document, *types.Array, types.Binary, bool, time.Time, types.NullType, types.Regex, types.Timestamp:
-				// type not supported for pushdown
-				// TODO $eq and $ne https://github.com/FerretDB/FerretDB/issues/1840
-				// TODO $gt and $lt https://github.com/FerretDB/FerretDB/issues/1875
-			case float64, string, types.ObjectID, int32, int64:
-				// Select if value under the key is equal to provided value.
-				sql := `(_jsonb->%[1]s)::jsonb = %[2]s`
-
-				// placeholder p.Next() returns SQL argument references such as $1, $2 to prevent SQL injections.
-				// placeholder $1 is used for field key,
-				// placeholder $2 is used for field value v.
-				filters = append(filters, fmt.Sprintf(sql, p.Next(), p.Next()))
-				args = append(args, k, string(must.NotFail(pjson.MarshalSingleValue(v))))
-
-			default:
-				panic(fmt.Sprintf("Unexpected type of value: %v", v))
+		switch v := v.(type) {
+		case *types.Document:
+			iter := v.Iterator()
```

Comment:
This part becomes difficult to read. I think multiple level of switch makes it hard to follow.it was difficult to tell that documents were iterated for $eq and $ne

Reply:
That's why we need to split this function. Apart from separating builder logic, I would suggest also to keep the switch from `*types.Document` case inside another function, that would handle all operators. Wdyt?

Reply:
Will we enable pushdown for different operators with different types? For `$eq` enable support for `time.Time` and for `$ne` or another operator `types.ObjectID`?
If that's not the case we can put that type checks into a small helper function like `isPushdownEnabledForValue(v any) bool`.

Reply:
I believe, that for different filter types, we might send different queries. If we are talking about `$eq`: after enabling `*types.Array`, we will send different queries than for strings or numbers.

Reply:
That's unfortunate. 😞

Reply:
`isPushdownEnabledForOperatorValue(op ,v any) bool` would work though, but let's not do it in this issue. Next operator maybe